src/views/TodoView.vue

```diff
@@ -0,0 +1,329 @@
+<script setup lang="ts">
+import TableItem from "../components/TableItem.vue";
+import { computed, onMounted, ref } from "vue";
+import { ResponseData, TaskInfo, debounce } from "../util/shared";
+import {
+  deletePost,
+  getPosts,
+  postPost,
+  updatePost,
+} from "../util/HttpRequests";
+import FlexCard from "../components/FlexCard.vue";
+import FlexList from "../components/FlexList.vue";
+import FlexText from "../components/FlexText.vue";
+import AddTodo from "../components/AddTodo.vue"
+import { useRouter } from "vue-router";
+import IconButton from "../components/IconButton.vue";
+
+const DEBOUNCE_TIME = 250;
+
+let oldGet: ResponseData[]; // Keep a backup to restore from
+const router = useRouter()
+
+const sortingOrder = ref<SORT_OPTIONS>("task_dsc");
+
+const todoData = ref<ResponseData[]>([]);
+
+const filterInput = ref("");
+const addText = ref("");
+
+type SORT_OPTIONS = "task_asc" | "task_dsc" | "complete" | "incomplete";
+
+const sortingOptions: { text: string; value: SORT_OPTIONS }[] = [
+  { text: "Tasks (dsc)", value: "task_dsc" },
+  { text: "Tasks (asc)", value: "task_asc" },
+  { text: "Complete", value: "complete" },
+  { text: "Incomplete", value: "incomplete" },
+];
+
+const todoFiltered = computed(() => {
+  return [...todoData.value].filter((value) => {
+    // Copy or we end up blowing the stack
+    return value.todoName
+      .toLowerCase()
+      .includes(filterInput.value.toLowerCase());
+  });
+});
+
+
+const todoSorted = computed(() => {
+
+  const taskSort = (a: ResponseData, b: ResponseData) => {
+    return a.todoName.localeCompare(b.todoName); // Thanks Ethan!
+  };
+
+  const completeSort = (a: ResponseData, b: ResponseData) => {
+    return ~~b.isComplete - ~~a.isComplete;
+  };
+
+  // Required so vue updates.
+  // Sort / reverse modify the original array
+  // which doesn't register as a change!
+  const sorting = [...todoFiltered.value];
```

Comment:
`toSorted` fixes this issue but you might have to modify your ts config to target a higher JS version for it to recognize it as a valid method